install-jellyfin.ps1

```diff
@@ -0,0 +1,385 @@
+   clear-host
+   }
+else
+   {
+   # We are not running "as Administrator" - so relaunch as administrator
+   
+   # Create a new process object that starts PowerShell
+   $newProcess = new-object System.Diagnostics.ProcessStartInfo "PowerShell";
+   
+   # Specify the current script path and name as a parameter
+   $newProcess.Arguments = $myInvocation.MyCommand.Definition;
+   
+   # Indicate that the process should be elevated
+   $newProcess.Verb = "runas";
+   
+   # Start the new process
+   [System.Diagnostics.Process]::Start($newProcess);
+   
+   # Exit from the current, unelevated, process
+   exit
+   }
+
+if($Quiet.IsPresent -or $Quiet -eq $true){
+    $Script:JellyfinDataDir = "$env:AppData\jellyfin\"
+    if([string]::IsNullOrEmpty($InstallLocation)){$Script:DefaultJellyFinInstallDirectory = "$env:Appdata\jellyfin\"}else{$Script:DefaultJellyFinInstallDirectory = $InstallLocation}
+    
+    if([string]::IsNullOrEmpty($EmbyLibraryLocation)){$Script:defaultEmbyDataDir = "$env:Appdata\Emby-Server\data\"}else{$Script:defaultEmbyDataDir = $EmbyLibraryLocation}
+    
+    if($InstallAsService.IsPresent -or $InstallAsService -eq $true){$Script:InstallAsService = $true ; $Script:JellyfinDataDir = "C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\jellyfin\" }else{$Script:InstallAsService = $false}
+    if($null -eq $ServiceUser){$Script:InstallServiceAsUser = $false}else{$Script:InstallServiceAsUser = $true; $Script:UserCredentials = $ServiceUser;  $Script:JellyfinDataDir = "C:\Users\$($Script:UserCredentials.UserName)\Appdata\Roaming\jellyfin\"}
+    if($CreateDesktopShorcut.IsPresent -or $CreateDesktopShorcut -eq $true) {$Script:CreateShortcut = $true}else{$Script:CreateShortcut = $false}
+    if($MigrateEmbyLibrary.IsPresent -or $MigrateEmbyLibrary -eq $true){$Script:MigrateLibrary = $true}else{$Script:MigrateLibrary = $false}
+    if($LaunchJellyfin.IsPresent -or $LaunchJellyfin -eq $true){$Script:StartJellyfin = $true}else{$Script:StartJellyfin = $false}
+    
+    if(-not (Test-Path $Script:DefaultJellyFinInstallDirectory)){
+        mkdir $Script:DefaultJellyFinInstallDirectory
+    }
+    Copy-Item -Path $PSScriptRoot/* -DestinationPath "$Script:DefaultJellyFinInstallDirectory/" -Force -Recurse
+    if($Script:InstallAsService){
+        if($Script:InstallServiceAsUser){
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" install Jellyfin `"$Script:DefaultJellyFinInstallDirectory\jellyfin.exe`"
+            Start-Sleep -Milliseconds 500
+            &sc.exe config Jellyfin obj=".\$($Script:UserCredentials.UserName)" password="$($Script:UserCredentials.GetNetworkCredential().Password)"
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START 
+        }else{
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" install Jellyfin `"$Script:DefaultJellyFinInstallDirectory\jellyfin.exe`"
+            Start-Sleep -Milliseconds 500
```

Comment:
What is going on in this section?.. Why sleep and then sleep again? :)

Reply:
NSSM's service control actions sometimes return before the service can take another. It prevents an error on manipulating the service post install. Its not an elegant solution, but nothing in this script is :(

Reply:
My point was, there's sleep, then comment, then sleep again... Doesn't sound right :)

Reply:
If that's not a comment but some powershell magic then please forgive my ignorance :)